tpkg/credns-setup/Makefile.in

```diff
@@ -0,0 +1,236 @@
+SLAVE_CONFIG	= $(SLAVE)/$(SLAVE).cf
+SLAVE_IP	= @SLAVE_IP@
+SLAVE_PORT	= @SLAVE_PORT@
+
+SERVERS		= $(MASTER) $(PROXY) $(SLAVE)
+CONFIGS		= $(MASTER_CONFIG) $(PROXY_CONFIG) $(SLAVE_CONFIG)
+CONFIG_FILES	= @CONFIG_FILES@
+ZONES		= @ZONES@
+ZONE_FILES	= @ZONE_FILES@
+SIGNED_ZONE_FILES	= @SIGNED_ZONE_FILES@
+INCR_ZONES	= @INCR_ZONES@
+RUIN_ZONES	= @RUIN_ZONES@
+
+NSDC		= @NSDC@
+NSDC_SETUP	= @NSDC_SETUP@
+LDNS_READ_ZONE	= @LDNS_READ_ZONE@
+LDNS_KEYGEN	= @LDNS_KEYGEN@
+LDNS_KEYGEN_OPTIONS	= @LDNS_KEYGEN_OPTIONS@
+LDNS_SIGNZONE	= @LDNS_SIGNZONE@
+MULTITAIL	= @MULTITAIL@
+
+DNSKEY_ALGORITHM	= RSASHA256
+DNSKEYS		= @DNSKEYS@
+
+TSIG_ALGORITHM	= hmac-sha256
+TSIG_SIZE	= 256
+TSIGS		= @TSIGS@
+
+all: $(MASTER_DB) $(TSIGS)
+
+$(ZONE_FILES) $(MASTER_CONFIG) $(PROXY_CONFIG) $(SLAVE_CONFIG):
+	./config.status
+
+$(DNSKEYS):
+	@echo "-- Creating $@ ..."
+	@$(LDNS_KEYGEN) -a $(DNSKEY_ALGORITHM) $(LDNS_KEYGEN_OPTIONS) \
+    `echo "$(@:.private=)" | sed "s/^$(MASTER)\///g"` > .key
+	@for ext in .ds .key .private ; \
+    do  [ -f `cat .key`$$ext ] && mv `cat .key`$$ext $(MASTER) ; \
+    ln -s `cat .key`$$ext $(@:.private=)$$ext; \
+    done
+	@rm .key
+
+$(SIGNED_ZONE_FILES):
+	@echo "-- Signing zone `echo $(@:.signed=)|sed 's/^$(MASTER)\///g'` ..."
+	@$(LDNS_SIGNZONE) $(@:.signed=) $(@:.signed=)
+
+$(TSIGS):
+	@mkdir -p tsigs
+	@echo "-- Creating $@ ..."
+	@echo "key:" > $@
+	@echo "	name: `echo $(@:.tsig=)|sed 's/^tsigs\///g'`." >> $@
+	@echo "	algorithm: $(TSIG_ALGORITHM)" >> $@
+	@$(LDNS_KEYGEN) -a $(TSIG_ALGORITHM) -b $(TSIG_SIZE) \
+    $(LDNS_KEYGEN_OPTIONS) `echo $(@:.tsig=)|sed 's/^tsigs\///g'` > .keyname
+	@grep "^Key:" `cat .keyname`.private \
+    | sed -e 's/^Key: /	secret: "/g' -e 's/$$/"/g' >> $@
+	@rm `cat .keyname`.key
+	@rm `cat .keyname`.private
+	@rm .keyname
+
+$(MASTER)/nsd_control.key:
+	@$(NSDC_SETUP) -d $(MASTER)
+$(PROXY)/nsd_control.key:
+	@$(NSDC_SETUP) -d $(PROXY)
+$(SLAVE)/nsd_control.key:
+	@$(NSDC_SETUP) -d $(SLAVE)
+
+start-$(MASTER) $(MASTER)/nsd.pid: $(SIGNED_ZONE_FILES) $(TSIGS) $(MASTER)/nsd_control.key
+	@( $(NSDC) -c $(MASTER_CONFIG) start \
+	&& echo "-- Started $(MASTER) ..."   ) \
+     || (  $(NSDC) -c $(MASTER_CONFIG) reload \
+	&& echo "-- Restarted $(MASTER) ..."   ) \
+     || echo "-- Could not (re)start $(MASTER)"
+start-$(PROXY) $(PROXY)/nsd.pid: $(TSIGS) $(PROXY)/nsd_control.key
+	@( $(NSDC) -c $(PROXY_CONFIG) start \
+	&& echo "-- Started $(PROXY) ..."   ) \
+     || (  $(NSDC) -c $(PROXY_CONFIG) reload \
+	&& echo "-- Restarted $(PROXY) ..."   ) \
+     || echo "-- Could not (re)start $(PROXY)"
+start-$(SLAVE) $(SLAVE)/nsd.pid: $(TSIGS) $(SLAVE)/nsd_control.key
+	@( $(NSDC) -c $(SLAVE_CONFIG) start \
+	&& echo "-- Started $(SLAVE) ..."   ) \
+     || (  $(NSDC) -c $(SLAVE_CONFIG) reload \
+	&& echo "-- Restarted $(SLAVE) ..."   ) \
+     || echo "-- Could not (re)start $(SLAVE)"
+
+start: start-$(MASTER) start-$(PROXY) start-$(SLAVE)
+restart: restart-$(MASTER) restart-$(PROXY) restart-$(SLAVE)
```

Comment:
The restart-* commands do not exist. The start-* ones are defined, also with 'restart' echo, but the line 'restart: ...' is not operative.  I guess you could remove it if it is not used.

Reply:
Removed it

Reply:
Okay, (I think the restart, restart-.. are also mentioned in the readme for this section, by the way).

Reply:
Done